server/services/notify/notifymentions/mentions_backend.go

```diff
@@ -141,3 +159,79 @@ func safeCallListener(listener MentionListener, userID string, evt notify.BlockC
 	}()
 	listener.OnMention(userID, evt)
 }
+
+func (b *Backend) deliverMentionNotification(username string, extract string, evt notify.BlockChangeEvent) (string, error) {
+	mentionedUser, err := b.delivery.UserByUsername(username)
+	if err != nil {
+		if b.delivery.IsErrNotFound(err) {
+			// not really an error; could just be someone typed "@sometext"
+			return "", nil
+		} else {
+			return "", fmt.Errorf("cannot lookup mentioned user: %w", err)
+		}
+	}
+
+	if evt.ModifiedBy == nil {
+		return "", fmt.Errorf("invalid user cannot mention: %w", ErrMentionPermission)
+	}
+
+	if evt.Board.Type == model.BoardTypeOpen {
+		// public board rules:
+		//    - admin, editor, commenter: can mention anyone on team (mentioned users are automatically added to board)
+		//    - guest: can mention board members
+		switch {
+		case evt.ModifiedBy.SchemeAdmin, evt.ModifiedBy.SchemeEditor, evt.ModifiedBy.SchemeCommenter:
+			if !b.permissions.HasPermissionToTeam(mentionedUser.Id, evt.TeamID, model.PermissionViewTeam) {
+				return "", fmt.Errorf("%s cannot mention non-team member %s : %w", evt.ModifiedBy.UserID, mentionedUser.Id, ErrMentionPermission)
+			}
+			// add mentioned user to board (if not already a member)
+			member, err := b.store.GetMemberForBoard(evt.Board.ID, mentionedUser.Id)
+			if member == nil || b.store.IsErrNotFound(err) {
```

Comment:
Should this be an && check? Assuming member is nil on any error, then only the member check would be necessary.

Reply:
Our store APIs are inconsistent in returning an error when a lookup has zero records found.  Sometimes an ErrNotFound is returned with nil pointer, sometimes a nil error and nil pointer.  Also some store APIs return `store.ErrNotFound` and others return `sql.ErrNoRows`. The latter doesn't match `store.IsErrNotFound` which is the problem in this case.  Once the tech debt card to clean this up is complete the code above will still work, but also works now.